pkgs/desktops/lomiri/development/geonames/default.nix

```diff
@@ -0,0 +1,117 @@
+{ stdenv
+, lib
+, fetchFromGitLab
+, gitUpdater
+, testers
+, buildPackages
+, cmake
+, docbook-xsl-nons
+, docbook_xml_dtd_45
+, gettext
+, glib
+, glibcLocales
+, withExamples ? true
+, gtk3
+# Uses gtkdoc-scan* tools, which produces a binary linked against lib for hostPlatform and executes it to generate docs
+, withDocumentation ? stdenv.buildPlatform.canExecute stdenv.hostPlatform
+, gtk-doc
+, pkg-config
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "geonames";
+  version = "0.3.0";
+
+  src = fetchFromGitLab {
+    owner = "ubports";
+    repo = "development/core/geonames";
+    rev = finalAttrs.version;
+    hash = "sha256-Mo7Khj2pgdJ9kT3npFXnh1WTSsY/B1egWTccbAXFNY8=";
+  };
+
+  outputs = [
+    "out"
+    "dev"
+  ] ++ lib.optionals withExamples [
+    "bin"
+  ] ++ lib.optionals withDocumentation [
+    "doc"
+  ];
+
+  postPatch = ''
+    patchShebangs src/generate-locales.sh tests/setup-test-env.sh
+
+    substituteInPlace doc/reference/CMakeLists.txt \
+      --replace "\''${CMAKE_INSTALL_DATADIR}/gtk-doc/html/\''${PROJECT_NAME}" "\''${CMAKE_INSTALL_DOCDIR}"
+    substituteInPlace demo/CMakeLists.txt \
+      --replace 'RUNTIME DESTINATION bin' 'RUNTIME DESTINATION ''${CMAKE_INSTALL_BINDIR}'
+  '' + lib.optionalString (stdenv.buildPlatform != stdenv.hostPlatform) ''
+    # Built for hostPlatform, executed during build
+    substituteInPlace src/CMakeLists.txt \
+      --replace 'COMMAND mkdb' 'COMMAND ${stdenv.hostPlatform.emulator buildPackages} mkdb'
```

Comment:
Is this upstreamable?

Reply:
I don't see how this could be upstreamed. Ideally upstream shouldn't rely on the user having an emulator to run the build-time executable - building for the native system (`buildPlatform`) in a cross situation with CMake is abit tricky but IMO the real solution here.

We could use `CMAKE_CROSSCOMPILING_EMULATOR=${stdenv.hostPlatform.emulator buildPackages}` instead of patching in our emulator call, I guess that would be slightly better, but still not a real fix.

Reply:
Switched to setting `CMAKE_CROSSCOMPILING_EMULATOR`, confirmed that it works by cross-compiling for armv7l-linux.

Ideally upstream will address this with a proper fix Soon:tm:, either for or after the 22.04 release of ubtouch.

Reply:
things like `CMAKE_INSTALL_DOCDIR`/`CMAKE_INSTALL_BINDIR` should be upstreamable

Reply:
Ah right, that stuff is upstreamable: https://gitlab.com/ubports/development/core/geonames/-/merge_requests/3